OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -71,7 +71,12 @@ public override void OnBecomingIdle(Actor self)
 			if (withSpriteBody.DefaultAnimation.CurrentSequence.Name != Info.IdleSequence)
 				withSpriteBody.DefaultAnimation.PlayRepeating(Info.IdleSequence);
 
-			base.OnBecomingIdle(self);
+			var thumper = self.World.FindActorsInCircle(self.CenterPosition, Info.MaxSearchRadius)
```

Comment:
You could make the target selection more openra-ey by removing the `x.HasTrait<AttractsWorms>()` check and using target types exclusively:

```csharp
// If close enough, we don't care about other actors.
var target = self.World.FindActorsInCircle(self.CenterPosition, Info.IgnoreNoiseAttackRange)
	.FirstOrDefault(x => attackTrait.HasAnyValidWeapons(Target.FromActor(x)));
```